tests/integration/test_take.py

```diff
@@ -64,64 +64,220 @@ def test_with_out_array():
     shape = (5,) * ndim
     np_arr = mk_seq_array(np, shape)
     num_arr = mk_seq_array(num, shape)
-    np_indices = mk_seq_array(np, (4,))
-    num_indices = mk_seq_array(num, (4,))
+    np_indices = mk_seq_array(np, shape_in)
+    num_indices = mk_seq_array(num, shape_in)
+
     res_np = np.take(np_arr, np_indices)
     res_num = num.take(num_arr, num_indices)
-
     assert np.array_equal(res_num, res_np)
+
     for axis in range(ndim):
         res_np = np.take(np_arr, np_indices, axis=axis)
         res_num = num.take(num_arr, num_indices, axis=axis)
         assert np.array_equal(res_num, res_np)
 
-    np_indices = mk_seq_array(np, (8,))
-    num_indices = mk_seq_array(num, (8,))
+
+@pytest.mark.parametrize("mode", ("clip", "wrap"))
+@pytest.mark.parametrize(
+    "shape_in",
+    ((8,), pytest.param((3, 4), marks=pytest.mark.xfail)),
+    ids=lambda shape_in: f"(shape_in={shape_in})",
+)
+@pytest.mark.parametrize("ndim", range(1, LEGATE_MAX_DIM + 1))
+def test_ndim_mode(ndim, mode, shape_in):
+    # for shape_in=(3, 4) and ndim=4,
+    # In Numpy, pass
+    # In cuNumeric, it raises ValueError:
+    # Point cannot exceed 4 dimensions set from LEGATE_MAX_DIM
+    shape = (5,) * ndim
+    np_arr = mk_seq_array(np, shape)
+    num_arr = mk_seq_array(num, shape)
+    np_indices = mk_seq_array(np, shape_in)
+    num_indices = mk_seq_array(num, shape_in)
 
     res_np = np.take(np_arr, np_indices, mode=mode)
     res_num = num.take(num_arr, num_indices, mode=mode)
-
     assert np.array_equal(res_num, res_np)
+
     for axis in range(ndim):
         res_np = np.take(np_arr, np_indices, axis=axis, mode=mode)
         res_num = num.take(num_arr, num_indices, axis=axis, mode=mode)
         assert np.array_equal(res_num, res_np)
 
 
+class TestTakeErrors:
+    def setup_method(self):
+        self.A_np = mk_seq_array(np, (3, 4, 5))
+        self.A_num = mk_seq_array(num, (3, 4, 5))
+
+    @pytest.mark.parametrize(
+        "indices",
+        (-5, 4),
+        ids=lambda indices: f"(indices={indices})",
+    )
+    def test_indices_invalid_scalar(self, indices):
+        expected_exc = IndexError
+        axis = 1
+        with pytest.raises(expected_exc):
+            np.take(self.A_np, indices, axis=axis)
+        with pytest.raises(expected_exc):
+            num.take(self.A_num, indices, axis=axis)
+
+    @pytest.mark.parametrize(
+        "indices",
+        ([-5, 0, 2], [0, 4, 2]),
+        ids=lambda indices: f"(indices={indices})",
+    )
+    def test_indices_invalid_array(self, indices):
+        expected_exc = IndexError
+        axis = 1
+        with pytest.raises(expected_exc):
+            np.take(self.A_np, np.array(indices), axis=axis)
+        with pytest.raises(expected_exc):
+            num.take(self.A_num, num.array(indices), axis=axis)
+
+    def test_invalid_indices_for_empty_array(self):
+        expected_exc = IndexError
+        # A_np = mk_seq_array(np, (3, 0))
+        # A_num = mk_seq_array(num, (3, 0))
```

Comment:
commented code should be deleted, generally (here and below) is there are reason to keep this here?

Reply:
Sorry, it's a typo. I have fixed it. Thanks.